### ISSUE TITLE
fix(model): keep absent eos_token_id from becoming token 0

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12604]
# fix(model): keep absent eos_token_id from becoming token 0

```routing
lane: model
paths: ["internal/model/config.go", "internal/model/audit_missing_eos_test.go"]
expected_steps: 4
```

## Current state

At source SHA `6872f998b465d96633c3356ef2382c97fc98cc7c`, a Hugging Face config with no `eos_token_id` leaves `Config.EOSTokenID` at Go's zero value. `Config.IsEOS(0)` then returns true because the EOS list is empty and the scalar fallback is token `0`.

This makes token `0` a stop token even when the model config did not define one. The audit witness also forces all-zero logits, so greedy decode chooses token `0` and `Generate` stops after one token instead of producing the requested fixed-length decode.

Duplicate search status: local ticket/docs search and parent GitHub search found no exact existing issue for absent EOS metadata defaulting to token `0` or for the resulting false stop.

Relevant seams:

- `internal/model/config.go:489-493` (`Config.UnmarshalJSON`) assigns `EOSTokenIDs` from the parsed auxiliary EOS field and only assigns `EOSTokenID` when the list is non-empty; absent EOS leaves the scalar at zero.
- `internal/model/config.go:1131-1142` (`Config.IsEOS`) treats an empty EOS list as "use scalar `EOSTokenID`".
- `internal/model/kv.go:1291-1296` (`Session.Generate`) breaks when `s.M.Cfg.IsEOS(next)` returns true.

## Working spine

Make absent EOS metadata enter the same no-early-stop state already used by fixed-length decode, then pin both config and generation behavior with tests.

## Parent context

Independent inference-correctness audit leaf from the 2026-09-09 model config/generation review.

## Why this is next

Absent EOS metadata currently fabricates token `0` as a stop token, so fixed-length generation can terminate early on a valid vocabulary id.

## Core through-line

Missing EOS metadata should mean "no early stop" unless a caller explicitly sets an EOS token. Token `0` is a valid vocabulary id and must not be fabricated as a stop token by config decoding.

## In scope

- `Config.UnmarshalJSON` behavior when `eos_token_id` is absent.
- `Config.IsEOS(0)` for absent EOS metadata.
- A generation regression proving token `0` is not treated as a fabricated stop token.

## Gold-plating boundary

Do not change tokenizer handling, add generation policy flags, or reinterpret configs that already contain scalar or list EOS ids. This ticket only covers absent `eos_token_id` preserving the existing `EOSTokenID == -1` "never early-stop" convention.

## Out of scope

Tokenizer-level stop policy, prompt templates, and generation API redesign are out of scope.

## Implementation steps

1. Update `Config.UnmarshalJSON` so absent EOS metadata leaves the config in the existing no-early-stop state, such as `EOSTokenID: -1` with an empty EOS list.
2. Preserve current behavior when scalar or list `eos_token_id` is present.
3. Add `internal/model/audit_missing_eos_test.go` with one direct `Config.IsEOS(0)` regression and one generation false-stop regression.
4. Run the witness and the existing config/EOS tests to confirm token `0` is only EOS when explicitly configured.

## Done condition

- [ ] JSON configs with no `eos_token_id` decode to a state where `Config.IsEOS(0)` is false.
- [ ] Existing scalar `eos_token_id` and list `eos_token_id` behavior remains unchanged.
- [ ] A generation witness with absent EOS and all-zero logits returns the requested number of tokens instead of stopping after token `0`.
- [ ] Tests cover both the direct `Config.IsEOS(0)` contract and the generation false-stop regression.

## Definition of Done

Done when absent EOS metadata maps to the existing no-early-stop convention, configured scalar/list EOS behavior is unchanged, and the two missing-EOS witnesses pass.

## Acceptance gate

`go test ./internal/model -run 'TestAuditMissingEOS(DoesNotDefaultToTokenZero|FalseStopsGeneration)' -count=1` passes after the fix, and existing config EOS tests still pass.

## Closure binding

Resolving commit cites the created GitHub issue and carries `(fak model)`.

## Witness

Captured failure from the audit overlay:

```text
$ go test ./internal/model -run 'TestAuditMissingEOS(DoesNotDefaultToTokenZero|FalseStopsGeneration)' -count=1
--- FAIL: TestAuditMissingEOSDoesNotDefaultToTokenZero (0.00s)
    audit_repro_test.go:132: config without eos_token_id treats token 0 as EOS: EOSTokenID=0 EOSTokenIDs=[]
--- FAIL: TestAuditMissingEOSFalseStopsGeneration (0.00s)
    audit_repro_test.go:165: missing eos_token_id defaulted to token 0 and stopped generation early: got [0], want three decoded tokens
FAIL
FAIL    github.com/anthony-chaudhary/fak/internal/model    0.024s
```

## Verifiable Witness

Place this standalone test in `internal/model/audit_missing_eos_test.go`, then run:

```bash
go test ./internal/model -run 'TestAuditMissingEOS(DoesNotDefaultToTokenZero|FalseStopsGeneration)' -count=1
```

```go
package model

import (
	"encoding/binary"
	"encoding/json"
	"math"
	"strings"
	"testing"
)

func auditConfigWithoutEOS(t *testing.T) Config {
	t.Helper()
	var cfg Config
	if err := json.Unmarshal([]byte(`{
		"model_type":"llama",
		"vocab_size":4,
		"hidden_size":32,
		"intermediate_size":64,
		"num_hidden_layers":1,
		"num_attention_heads":2,
		"num_key_value_heads":2,
		"head_dim":16,
		"rms_norm_eps":1e-6,
		"rope_theta":10000,
		"tie_word_embeddings":true
	}`), &cfg); err != nil {
		t.Fatal(err)
	}
	return cfg
}

func TestAuditMissingEOSDoesNotDefaultToTokenZero(t *testing.T) {
	cfg := auditConfigWithoutEOS(t)
	if cfg.IsEOS(0) {
		t.Fatalf("config without eos_token_id treats token 0 as EOS: EOSTokenID=%d EOSTokenIDs=%v", cfg.EOSTokenID, cfg.EOSTokenIDs)
	}
}

func TestAuditMissingEOSFalseStopsGeneration(t *testing.T) {
	cfg := auditConfigWithoutEOS(t)
	m := NewSynthetic(cfg)
	for name, meta := range m.manifest {
		raw := m.raw[meta.Offset : meta.Offset+meta.Nbytes]
		fill := uint32(0)
		if strings.HasSuffix(name, "layernorm.weight") || name == "model.norm.weight" {
			fill = math.Float32bits(1)
		}
		for off := 0; off < len(raw); off += 4 {
			binary.LittleEndian.PutUint32(raw[off:], fill)
		}
	}
	got := m.NewSession().Generate([]int{1}, 3)
	if len(got) != 3 {
		t.Fatalf("missing eos_token_id defaulted to token 0 and stopped generation early: got %v, want three decoded tokens", got)
	}
	for _, id := range got {
		if id != 0 {
			t.Fatalf("setup expected all-zero logits to choose token 0, got %v", got)
		}
	}
}
```

## Likely files

- `internal/model/config.go`
- `internal/model/audit_missing_eos_test.go`

## Lane

model

## Work unit

leaf

## Expected steps

4

## Trigger

Inference audit witness captured on 2026-09-09.

## Batch policy

One GitHub issue per `fak-model-key`; update this issue if another absent-EOS/token-zero false-stop reproduction lands for the same config seam.

## Problem frame

- Centrality: Core
- P1 Context: preserved - concrete config and generation witness at source SHA `6872f998b465d96633c3356ef2382c97fc98cc7c`; direct seam is config decoding, not scheduler state.
- P2 Net value: advanced - prevents false early-stop behavior for model configs that omit EOS metadata.
- P3 Adaptation: preserved - keeps scalar and list EOS configs authoritative when present.
- P4 Operations: preserved - one package, two likely files, four implementation steps.

## Blast radius / fallback

Blast radius is limited to EOS parsing and `Config.IsEOS`. The fallback is existing tests for scalar EOS, list EOS, fixed-length decode with `EOSTokenID: -1`, and generation tests that intentionally stop on configured EOS.
